packages/node-resolve/src/index.js

```diff
@@ -63,13 +63,22 @@ export function nodeResolve(opts = {}) {
       options.dedupe.includes(importee) || options.dedupe.includes(getPackageName(importee));
   }
 
-  const resolveOnly = options.resolveOnly.map((pattern) => {
-    if (pattern instanceof RegExp) {
-      return pattern;
-    }
-    const normalized = pattern.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
-    return new RegExp(`^${normalized}$`);
-  });
+  // creates a function from the patterns to test if a particular module should be bundled.
+  const allowPatterns = (patterns) => {
+    const regexPatterns = patterns.map((pattern) => {
+      if (pattern instanceof RegExp) {
+        return pattern;
+      }
+      const normalized = pattern.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+      return new RegExp(`^${normalized}$`);
+    });
+    return (id) => !regexPatterns.length || regexPatterns.some((pattern) => pattern.test(id));
```

Comment:
It is :) 

Removing the `!regexPatterns.length` causes tests to fail. I thought it was a bit backwards at first too, but I think it makes sense in the context of "zero elements in the array means there are no filters, and no filters means include everything"

Reply:
Cool. Hmm I see it as zero items in an allow list meaning there’s nothing to allow 🤷 

also follows the semantics of `[].some(…)` then too